src/Model/ModelManager.php

```diff
@@ -414,7 +415,14 @@ public function addIdentifiersToQuery($class, ProxyQueryInterface $queryProxy, a
             foreach ($fieldNames as $posName => $name) {
                 $parameterName = sprintf('field_%s_%s_%d', $prefix, $name, $pos);
                 $ands[] = sprintf('%s.%s = :%s', current($qb->getRootAliases()), $name, $parameterName);
-                $qb->setParameter($parameterName, $ids[$posName]);
+
```

Comment:
Can you please try to extract some small methods. This will improve readability.

Reply:
done